utils/src/lib.rs

```diff
@@ -636,6 +636,49 @@ fn format_param(param: &Param, structs: &mut HashSet<String>) -> String {
     }
 }
 
+fn format_event_params(param: &EventParam, structs: &mut HashSet<String>) -> String {
+    let (kind, v2_struct) = match param.kind {
```

Comment:
Can you refactor this to be reused across both functions and events to avoid code duplication?